code/modules/hivemind/mobs.dm

```diff
@@ -0,0 +1,916 @@
+/mob/living/simple_animal/hostile/hivemind/mechiver/Destroy()
+	release_passenger(TRUE)
+	..()
+
+/mob/living/simple_animal/hostile/hivemind/mechiver/death()
+	release_passenger(TRUE)
+	..()
+	gibs(loc, null, /obj/effect/gibspawner/robot)
+	if(pilot)
+		gibs(loc, null, /obj/effect/gibspawner/human)
+	qdel(src)
+
+
+/////////////////////////////////////PHASER///////////////////////////////////
+//Special ability: Superposition. Phaser exists at four locations. But, actually he vulnerable only at one. Other is just a copies
+//Moves with teleportation only, can stun victim if he land on it
+//Also can hide in closets
+//Can't speak, no malfunctions
+//Appears from dead human body
+//////////////////////////////////////////////////////////////////////////////
+
+/mob/living/simple_animal/hostile/hivemind/phaser
+	name = "phaser"
+	desc = "Crooked man with strange device on it's head. He twitches sometimes and... What are you looking on? Run!"
+	icon = 'icons/mob/hivemind.dmi'
+	icon_state = "phaser-1"
+	health = 120
+	maxHealth = 120
+	speak_chance = 0
+	malfunction_chance = 0
+	mob_size = MOB_MEDIUM
+	ability_cooldown = 2 MINUTES
+	//internals
+	var/can_use_special_ability = TRUE
+	var/list/my_copies = list()
+
+	New()
+		..()
+		filters += filter(type="blur", size = 0)
+
+
+/mob/living/simple_animal/hostile/hivemind/phaser/Life()
+	stop_automated_movement = TRUE
+	..()
+
+	//special ability using
+	if(world.time > special_ability_cooldown && can_use_special_ability)
+		if(target_mob && (health <= 50))
+			special_ability()
+
+	//closet hiding
+	if(!target_mob)
+		var/obj/structure/closet/C = locate() in get_turf(src)
+		if(C && loc != C)
+			if(!C.opened)
+				C.open(src)
+			if(C.opened)
+				C.close(src)
+		for(var/obj/structure/closet/Closet in view(src))
+			if(!Closet.locked && !Closet.welded)
+				phase_move_to(Closet)
+				break
+
+
+/mob/living/simple_animal/hostile/hivemind/phaser/AttackTarget()
+	if(target_mob && get_dist(src, target_mob) > 1)
+		stance = HOSTILE_STANCE_ATTACK
+	..()
+
+
+/mob/living/simple_animal/hostile/hivemind/phaser/MoveToTarget()
+	if(!target_mob || SA_attackable(target_mob))
+		stance = HOSTILE_STANCE_IDLE
+	if(target_mob in ListTargets(10))
+		if(get_dist(src, target_mob) > 1)
+			stance = HOSTILE_STANCE_ATTACK
+			phase_move_to(target_mob, nearby = TRUE)
+		else
+			stance = HOSTILE_STANCE_ATTACKING
+
+
+/mob/living/simple_animal/hostile/hivemind/phaser/proc/is_can_jump_on(var/turf/target)
+	if(!target || target.density || istype(target, /turf/space) || istype(target, /turf/simulated/open))
+		return FALSE
+
+	//to prevent reflection's stacking
+	var/mob/living/simple_animal/hostile/hivemind/phaser/P = locate() in target
+	if(P)
+		return FALSE
+
+	for(var/obj/O in target)
+		if(!O.CanPass(src, target))
+			return FALSE
+
+	return TRUE
+
+
+//first part of phase moving is just preparation
+/mob/living/simple_animal/hostile/hivemind/phaser/proc/phase_move_to(var/atom/target, var/nearby = FALSE)
```

Comment:
var/ isn't needed in proc definitions, don't think this is a contributing guideline though? not entirely sure.